### PR TITLE
Query: range query histogram

### PR DESCRIPTION
**Pull request by thorfour**
Adds a range query histogram that tracks the distribution
of range queries. This can be useful for determining storage
retention requirements.

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

- Adds a range query histogram to the query component.

### CODE REVIEW

File: pkg/api/query/v1.go [hunk 1]
```diff
@@ -430,6 +441,9 @@ func (qapi *QueryAPI) queryRange(r *http.Request) (interface{}, []error, *api.Ap
 
 	qe := qapi.queryEngine(maxSourceResolution)
 
+	// Record the query range requested
```
Comment (GiedriusS) [inline pkg/api/query/v1.go:444]: Minor nit: let's end comments with a full stop:
```suggestion
	// Record the query range requested.
```

**Review by GiedriusS — APPROVED** [review]

File: pkg/api/query/v1.go [hunk 2]
```diff
@@ -142,6 +147,12 @@ func NewQueryAPI(
 		defaultInstantQueryMaxSourceResolution: defaultInstantQueryMaxSourceResolution,
 		defaultMetadataTimeRange:               defaultMetadataTimeRange,
 		disableCORS:                            disableCORS,
+
+		queryRangeHist: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
+			Name:    "query_range_duration_seconds",
```
Comment (brancz) [inline pkg/api/query/v1.go:152]:
```suggestion
			Name:    "thanos_query_range_requested_timespan_duration_seconds",
```

To be consistent with the rest of the Thanos metrics this should have the `thanos_` prefix, I would also suggest we make this metric slightly more descriptive.

**Review by brancz — COMMENTED** [review]
Just a small nit on naming, otherwise this is great!

File: pkg/api/query/v1.go (hunk 2 above)
Comment (thorfour) [reply pkg/api/query/v1.go]: Done.

**Review by brancz — APPROVED** [review]

**Pull request merged**